### PR TITLE
made host analyzer not to load assembly to find out analyzer referenc…

### PR DESCRIPTION
**Pull request by heejaechang**
…e dll location

unlike analyzers through nuget which is per project and on demand, host analyzer is context less so it will load dlls that belong to any language such as typescript, xaml, F# and more blindly. this change should prevent it from happening.

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @333fred @tmat @ivanbasov @jinujoseph can you take a look? this should fix Rps regression.

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest please

**Comment by heejaechang** [comment]
retest windows_release_vs-integration_prtest please

File: src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs [hunk 1]
```diff
@@ -159,6 +206,50 @@ public AnalyzerReference ReadAnalyzerReferenceFrom(ObjectReader reader, Cancella
             throw ExceptionUtilities.UnexpectedValue(type);
         }
 
+        private Checksum CreateAnalyzerReferenceChecksum(AnalyzerReference reference, CancellationToken cancellationToken)
```
Comment (ivanbasov) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: What is the purpose of introducing CreateAnalyzerReferenceChecksum? The original CreateChecksum method becomes an almost transparent. Does CreateChecksum still add a value?

Comment (heejaechang) [inline src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs]: sure, let me merge two.

**Pull request merged**